macroid-core/src/main/scala/macroid/Contexts.scala

```diff
@@ -45,12 +46,19 @@ object ContextWrapper {
 
   def apply[F](fragment: F)(implicit fragmentImpl: Fragment[F]): ActivityContextWrapper =
     ContextWrapper(fragmentImpl.activity(fragment))
+
+  def apply(view: View): ContextWrapper = view.getContext match {
+    case activity: Activity => ContextWrapper(activity)
+    case service: Service => ContextWrapper(service)
+    case app: Application => ContextWrapper(app)

```

Comment:
The [Androd Documentation](http://developer.android.com/intl/es/reference/android/view/View.html#View(android.content.Context)) sais that this context through which it can access the current theme, resources, etc. 

In my experience, this context always is an `Activity` or `Service`, but maybe in the future the Android SDK can include other types of contexts, I think that no, but it's possible

Maybe we should create a new `ContextWrapper` for simple contexts. The name can be `DefaultContextWraper`. The code could be then

```scala
case class DefaultContextWrapper(original: WeakReference[Context], application: Context)
  extends ContextWrapper { type C = Context }
```
and a new apply in `ContextWraper` object

```scala
  def apply(view: View): ContextWrapper =
    DefaultContextWrapper(WeakReference(view.getContext), view.getContext.getApplicationContext)
```

In this case, we won't have problems

What's your opinion about that? I think that I prefer a `DefaultContextWrapper` instead of the pattern matching. Do you like `DefaultContextWrapper` name? :-)